hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -114,11 +126,15 @@ public class TransferPrecompile extends AbstractWritePrecompile {
     private final int functionId;
     private final Address senderAddress;
     private final ImpliedTransfersMarshal impliedTransfersMarshal;
+    private final AutoCreationLogic autoCreationLogic;
+    private final RecordsHistorian recordsHistorian;
     private ResponseCodeEnum impliedValidity;
     private ImpliedTransfers impliedTransfers;
     private List<BalanceChange> explicitChanges;
     private HederaTokenStore hederaTokenStore;
     protected CryptoTransferWrapper transferOp;
+    private final Set<AccountID> requestedLazyCreates = new HashSet<>();
```

Comment:
It seems like it might be cleaner to create an interface like,
```
interface CreationAwareAliasResolver {
    // Returns 0x00...00 for an unrecognized address unless mayBeImplicitCreation=true;
    // then it returns the given address
    byte[] resolve(byte[] address, boolean isPossibleImplicitCreation);
}
```
and then use it in the `decode*Transfers` methods, so that in `constructBalanceChanges()` [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/TransferPrecompile.java#L525) we can just construct an aliased `BalanceChange`s from any non-mirror address.

Reply:
Changed to some form of the suggested...

`unaliased(byte[])` method returns the address itself,  not  0x00..00 for unrecognized addresses, so I think checking that the returned address exists or not will be enough to discern what type of change we need 🤔